### PR TITLE
some simple PAM fixes split out of #15742

### PR DESCRIPTION
**Pull request by poettering**
Let's split out some simpler stuff from #15742 to track down what the CI woes are with it. I'd assume this PR should be fine to merge quickly.

### CODE REVIEW

File: src/login/pam_systemd.c [hunk 1]
```diff
@@ -957,11 +956,23 @@ _public_ PAM_EXTERN int pam_sm_close_session(
                 int argc, const char **argv) {
 
         const void *existing = NULL;
+        bool debug = false;
         const char *id;
         int r;
 
         assert(handle);
 
+        if (parse_argv(handle,
```
Comment (vcaputo) [inline src/login/pam_systemd.c:965]: I wonder if parse_argv() should be changed to return void, unless there are plans to return errors.

**Comment by vcaputo** [comment]
lgtm other than the parse_argv() simplification opportunity

File: src/login/pam_systemd.c (hunk 1 above)
Comment (poettering) [reply src/login/pam_systemd.c:965]: hmm, yeah, hmm. i thought about this, but didn't want to do that in this patch or PR. And it feels as if sooner or later the function might have reason to fail, so maybe no worth dropping it. Also, this should probably have in pam_systemd and pam_systemd_home the same (there's a similar function there). 

I'd say: something to consider for a later PR...

**Comment by poettering** [comment]
thanks for the review!

**Comment by keszybz** [comment]
LGTM too.

**Pull request merged**